### PR TITLE
d4xx: add recovery string to dfu device

### PR DESCRIPTION
**Pull request by dmipx**
Bump to version: 1.0.1.23

### CODE REVIEW

File: kernel/realsense/d4xx.c [hunk 1]
```diff
@@ -431,7 +431,7 @@ struct ds5_dfu_dev {
 	enum dfu_state dfu_state_flag;
 	unsigned char *dfu_msg;
 	u16 msg_write_once;
-	unsigned char init_v4l_f;
+	// unsigned char init_v4l_f;
```
Comment (Nir-Az) [inline kernel/realsense/d4xx.c:434]: Why do we want to keep this as a comment? and also the reference to it below?
Is this needed?

File: kernel/realsense/d4xx.c [hunk 2]
```diff
@@ -4824,18 +4824,31 @@ static ssize_t ds5_dfu_device_read(struct file *flip,
...
 
 	if (mutex_lock_interruptible(&state->lock))
 		return -ERESTARTSYS;
-	ret |= ds5_read(state, DS5_FW_VERSION, &fw_ver);
-	ret |= ds5_read(state, DS5_FW_BUILD, &fw_build);
-	if (ret < 0)
-		goto e_dfu_read_failed;
-	snprintf(msg, sizeof(msg) ,"DFU info: \tver:  %d.%d.%d.%d\n",
+	if (state->dfu_dev.dfu_state_flag == DS5_DFU_RECOVERY) {
+		ds5_write_with_check(state, 0x500c, 0x00);
```
Comment (Nir-Az) [inline kernel/realsense/d4xx.c:4834]: 1. What does 0x500c means?
Maybe we should keep it as a define with a proper name?
2. Are we entering DFU state here? what didn't happened before?

File: kernel/realsense/d4xx.c [hunk 3]
```diff
@@ -5044,10 +5058,10 @@ static int ds5_dfu_device_release(struct inode *inode, struct file *file)
 	state->dfu_dev.device_open_count--;
 	if (state->dfu_dev.dfu_state_flag != DS5_DFU_RECOVERY)
 		state->dfu_dev.dfu_state_flag = DS5_DFU_IDLE;
-	if (state->dfu_dev.dfu_state_flag == DS5_DFU_DONE
-			&& state->dfu_dev.init_v4l_f)
-		ds5_v4l_init(state->client, state);
-	state->dfu_dev.init_v4l_f = 0;
+	// if (state->dfu_dev.dfu_state_flag == DS5_DFU_DONE
```
Comment (Nir-Az) [inline kernel/realsense/d4xx.c:5061]: Why was this removed?
And remain commented?

File: kernel/realsense/d4xx.c [hunk 4]
```diff
@@ -5478,6 +5492,8 @@ static int ds5_probe(struct i2c_client *c, const struct i2c_device_id *id)
 	if (rec_state == 0x201) {
 		dev_info(&c->dev, "%s(): D4XX recovery state\n", __func__);
 		state->dfu_dev.dfu_state_flag = DS5_DFU_RECOVERY;
+		/* Override I2C drvdata */
+		i2c_set_clientdata(c, state);
```
Comment (Nir-Az) [inline kernel/realsense/d4xx.c:5496]: What does this function does?
Can you explain the relocation?

File: kernel/realsense/d4xx.c [hunk 5]
```diff
@@ -5530,10 +5544,15 @@ static int ds5_probe(struct i2c_client *c, const struct i2c_device_id *id)
 
 static int ds5_remove(struct i2c_client *c)
 {
+#ifdef CONFIG_VIDEO_D4XX_SERDES
```
Comment (Nir-Az) [inline kernel/realsense/d4xx.c:5547]: Why do we want it here and not after line 5555 where they use it?

Comment (dmipx) [reply kernel/realsense/d4xx.c:5547]: [ISO C90 forbids mixed declarations and code in C](https://www.kernel.org/doc/html/v5.10/process/programming-language.html)

File: kernel/realsense/d4xx.c (hunk 4 above)
Comment (dmipx) [reply kernel/realsense/d4xx.c:5496]: Used in remove function L5552

File: kernel/realsense/d4xx.c (hunk 2 above)
Comment (dmipx) [reply kernel/realsense/d4xx.c:4834]: 1. This used in ds5_dfu_detach L4803, in order to get properly unlocked registers to read  ds5_dfu_get_dev_info.
2. We verify if camera in DFU state we can read DFU version. If camera in operation mode we cannot enter DFU and restore to operational. There is no opcode to exit DFU.

File: kernel/realsense/d4xx.c (hunk 1 above)
Comment (dmipx) [reply kernel/realsense/d4xx.c:434]: We want to resolve that in different way, leave it as reminder.

**Review by Nir-Az — APPROVED** [review]
Tested and ready to merge?

**Review by Nir-Az — APPROVED** [review]
